mne/rank.py

```diff
@@ -74,18 +101,33 @@ def _estimate_rank_from_s(s, tol='auto'):
     tol : float | 'auto'
         Tolerance for singular values to consider non-zero in calculating the
         rank. Can be 'auto' to use the same thresholding as
-        ``scipy.linalg.orth``.
+        ``scipy.linalg.orth`` (assuming np.float64 datatype) adjusted
+        by a factor of 2.
 
     Returns
     -------
     rank : int
         The estimated rank.
     """
     if isinstance(tol, str):
-        if tol != 'auto':
-            raise ValueError('tol must be "auto" or float')
-        eps = np.finfo(float).eps
-        tol = len(s) * np.amax(s) * eps
+        if tol not in ('auto', 'float32'):
+            raise ValueError('tol must be "auto" or float, got %r' % (tol,))
```

Comment:
float or float32?

Reply:
It means an actual float, the `"float32"` is a hidden, hopefully temporary, expert-only option (we don't want users to use it)

Reply:
but above you test float32. I am confused.

Reply:
got itI Missed 'float32' was a string.